core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -56,6 +56,29 @@ const std::unique_ptr<Kokkos::Cuda> &Kokkos::Impl::cuda_get_deep_copy_space(
   return space;
 }
 
+namespace {
+
+auto get_failure_mode(cudaError_t error_code) {
+  using FailureMode =
+      Kokkos::Experimental::RawMemoryAllocationFailure::FailureMode;
+  switch (error_code) {
+    case cudaErrorMemoryAllocation: return FailureMode::OutOfMemoryError;
+    case cudaErrorInvalidValue: return FailureMode::InvalidAllocationSize;
+    default: return FailureMode::Unknown;
+  }
+}
+
+void throw_cuda_allocation_failure(size_t alloc_size, cudaError_t error_code,
+                                   std::string msg) {
+  msg += " returned error code \"";
+  msg += cudaGetErrorName(error_code);
+  msg += "\"";
+  Kokkos::Impl::throw_bad_alloc(alloc_size, std::align_val_t{1},
```

Comment:
```suggestion
  Kokkos::Impl::throw_bad_alloc(alloc_size, std::align_val_t{256},
```
According to https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#device-memory-accesses:
> Any address of a variable residing in global memory or returned by one of the memory allocation routines from the driver or runtime API is always aligned to at least 256 bytes.